ingestion/src/metadata/ingestion/source/dashboard/quicksight.py

```diff
@@ -0,0 +1,221 @@
+
+    @classmethod
+    def create(cls, config_dict, metadata_config: OpenMetadataConnection):
+        config = WorkflowSource.parse_obj(config_dict)
+        connection: QuickSightConnection = config.serviceConnection.__root__.config
+        if not isinstance(connection, QuickSightConnection):
+            raise InvalidSourceException(
+                f"Expected QuickSightConnection, but got {connection}"
+            )
+        return cls(config, metadata_config)
+
+    def get_dashboards_list(self) -> Optional[List[dict]]:
+        """
+        Get List of all dashboards
+        """
+        dashboard_ids = [
+            dashboard["DashboardId"]
+            for dashboard in self.quicksight.list_dashboards(
+                AwsAccountId=self.aws_account_id
+            )["DashboardSummaryList"]
+        ]
+        dashboards = [
+            self.quicksight.describe_dashboard(
+                AwsAccountId=self.aws_account_id, DashboardId=dashboard_id
+            )["Dashboard"]
+            for dashboard_id in dashboard_ids
+        ]
+        return dashboards
+
+    def get_dashboard_name(self, dashboard_details: dict) -> str:
+        """
+        Get Dashboard Name
+        """
+        return dashboard_details["Name"]
+
+    def get_dashboard_details(self, dashboard: dict) -> dict:
+        """
+        Get Dashboard Details
+        """
+        return dashboard
+
+    def yield_dashboard(
+        self, dashboard_details: dict
+    ) -> Iterable[CreateDashboardRequest]:
+        """
+        Method to Get Dashboard Entity
+        """
+        dashboard_url = self.quicksight.get_dashboard_embed_url(
+            AwsAccountId=self.aws_account_id,
+            DashboardId=dashboard_details["DashboardId"],
+            IdentityType="ANONYMOUS",
+        )["EmbedUrl"]
+
+        yield CreateDashboardRequest(
+            name=dashboard_details["DashboardId"],
+            dashboardUrl=dashboard_url,
+            displayName=dashboard_details["Name"],
+            description=dashboard_details["Version"].get("Description", ""),
+            charts=[
+                EntityReference(id=chart.id.__root__, type="chart")
+                for chart in self.context.charts
+            ],
+            service=EntityReference(
+                id=self.context.dashboard_service.id.__root__, type="dashboardService"
+            ),
+        )
+
+    def yield_dashboard_chart(
+        self, dashboard_details: Any
+    ) -> Optional[Iterable[CreateChartRequest]]:
+        """Get chart method
+
+        Args:
+            dashboard_details:
+        Returns:
+            Iterable[CreateChartRequest]
+        """
+        dashboard_url = self.quicksight.get_dashboard_embed_url(
+            AwsAccountId=self.aws_account_id,
+            DashboardId=dashboard_details["DashboardId"],
+            IdentityType="ANONYMOUS",
+        )["EmbedUrl"]
+        charts = dashboard_details["Sheets"]
```

Comment:
I see a lot of key accesses to the data, e.g., `dashboard_details["Sheets"]`. Are all of these always going to be present? Maybe random question as I have never player with Quicksight, but could we have a dashboard without any Charts/Sheets?

It would be interesting to think about these scenarios and handle them. I am thinking of something like:

```python
charts = dashboard_details.get("Sheets")
for chart in charts or []:
    ...
```

This is just an example, feel free to tune it as needed and apply the same logic to wherever else might make sense.

Reply:
@pmbrull According to the [documentation](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/quicksight.html#QuickSight.Client.describe_dashboard), a dashboard seems guaranteed to have at least one sheet (i.e. `charts`):

> A sheet , which is an object that contains a set of visuals that are viewed together on one page in Amazon QuickSight. Every analysis and dashboard contains at least one sheet.

Given this, I don't think it's possible to have a dashboard without any Charts/Sheets in Quicksight. However, if you think we should still handle such theoretical edge cases, then I can change the logic accordingly.

Reply:
Hi @michizhou awesome, if its covered by the docs then we can keep it 🚀 

Just left that as a general though but you know more about Quicksight than I do :) Maybe passing the doc link as a comment could be interesting though

Reply:
If you have the time to prep a quick test that would be great. Otherwise we can merge this in and assign a followup ticket on that to not leave this open for too long. Then we can merge this in. Thanks!

Reply:
Hi @pmbrull I am currently working on adding some unit tests to this PR today. If I don't have the tests ready by the end of today feel free to open an issue to tackle them later and merge this PR in.

Reply:
@pmbrull I have added some unit tests to the PR. Please let me know if additional changes are needed.